pennylane_cirq/cirq_device.py

```diff
@@ -170,6 +182,17 @@ def __init__(self, wires, shots, qubits=None):
         "Projector": CirqOperation(lambda: cirq.ProductState.projector),
     }
 
+    def supports_operation(self, operation):
+        # pylint: disable=missing-function-docstring
+        op = operation
+        if isinstance(op, type) and issubclass(op, Operation):
+            op = op.name
```

Comment:
Careful! `name` is a property, thus it cannot be used as a class method:
```pycon
>>> qml.ops.Prod.name
<property at 0x10fc6e400>
```

Reply:
Will operation ever be a type (a `Pow` class)? If this happens we can't know if the operation is supported or not, given that since it is not an instance we don't know what the base is.

Reply:
So I'm not quite sure when an type is passed.  `dev.stopping_condition` and `dev.check_validity` both use the string name.